From ce4208add3b49d440e0c418229774dd802ee6b87 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 31 Mar 2016 21:55:55 -0400
Subject: [PATCH] main: allow users to override QT_QPA_PLATFORMTHEME

commit 971baf2e225abc5a67bd0c149f4cbfdb0ef8c549 added
code to make Qt applications look like GNOME ones.

That makes it impossible for users to use custom
platform themes though.

This commit addresses the problem by only setting
QT_QPA_PLATFORMTHEME if it isn't already set.

https://bugzilla.gnome.org/show_bug.cgi?id=762681
---
 gnome-session/main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gnome-session/main.c b/gnome-session/main.c
index 9f3ca0f..615b4ce 100644
--- a/gnome-session/main.c
+++ b/gnome-session/main.c
@@ -415,7 +415,9 @@ main (int argc, char **argv)
 
         /* Tell Qt to try to make Qt applications look like gnome
          */
-        gsm_util_setenv ("QT_QPA_PLATFORMTHEME", "qgnomeplatform");
+        if (g_getenv ("QT_QPA_PLATFORMTHEME") == NULL) {
+                gsm_util_setenv ("QT_QPA_PLATFORMTHEME", "qgnomeplatform");
+        }
 
         /* hack to fix keyring until we can reorder things in 3.20
          * https://bugzilla.gnome.org/show_bug.cgi?id=738205
